chats/infra/database/memory.go

```diff
@@ -0,0 +1,83 @@
+package database
+
+import (
+	account "github.com/There-is-Go-alternative/GoMicroServices/account/domain"
+	chats "github.com/There-is-Go-alternative/GoMicroServices/chats/domain"
+)
+
+type MockDatabase struct {
```

Comment:
Pour éviter de trop jouer avec les index, ton type MockDatabase aurait pu utiliser des maps:

```suggestion
type MockDatabase struct {
	Chats        map[string]chats.Chat
	Messages map[string][]chats.Message
}
```
ou même:
```suggestion
type MockDatabase struct {
	Chats        map[string]chats.Chat
	Messages map[string]map[string]chats.Message
}
```